### PR TITLE
.NET Core 3

### PR DESCRIPTION
**Pull request by MicrosoftHealthService**
Updates package 'Microsoft.Extensions.Configuration.Json' to version '3.0.0'
Updates package 'Microsoft.Extensions.Http' to version '3.0.0'
Updates package 'Microsoft.Extensions.Logging' to version '3.0.0'
Updates package 'Microsoft.AspNetCore.Authentication.JwtBearer' to version '3.0.0'
Updates package 'Microsoft.Extensions.FileProviders.Embedded' to version '3.0.0'
Updates package 'Microsoft.Extensions.Options.ConfigurationExtensions' to version '3.0.0'
Updates package 'Microsoft.Extensions.Configuration.Binder' to version '3.0.0'
Updates package 'Microsoft.Extensions.Diagnostics.HealthChecks' to version '3.0.0'
Updates package 'Microsoft.Extensions.Logging.Abstractions' to version '3.0.0'
Updates package 'Microsoft.Extensions.Options' to version '3.0.0'
Updates package 'Microsoft.Extensions.DependencyInjection' to version '3.0.0'
Updates package 'Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions' to version '3.0.0'
Updates package 'Microsoft.Extensions.Configuration.AzureKeyVault' to version '3.0.0'
Updates package 'Microsoft.Extensions.Logging.ApplicationInsights' to version '2.11.0'
Updates package 'Microsoft.Extensions.Hosting.Abstractions' to version '3.0.0'
Updates package 'Microsoft.AspNetCore.TestHost' to version '3.0.0'

Addresses [AB#71813](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/71813) #797

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/JsonArrayPool.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Buffers;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Api.Features.Formatters
+{
+    internal class JsonArrayPool : IArrayPool<char>
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/JsonArrayPool.cs]: Might be nice to explain why this class exists. Looks like a pass-through wrapper.

Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/JsonArrayPool.cs]: Yep agree, I was still working through trying to get things running. The class we were using got marked as internal so its no longer available. This adapts the Json.net IArrayPool to the .NET ArrayPool

File: src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FhirXmlInputFormatter.cs [hunk 2]
```diff
@@ -47,34 +45,29 @@ protected override bool CanReadType(Type type)
...
 
-            var request = context.HttpContext.Request;
+            context.HttpContext.AllowSynchronousIO();
 
-            if (!request.Body.CanSeek)
-            {
-                request.EnableBuffering();
-                await request.Body.DrainAsync(context.HttpContext.RequestAborted);
-                request.Body.Seek(0L, SeekOrigin.Begin);
-            }
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FhirXmlInputFormatter.cs]: Undo this delete

File: global.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
   "sdk": {
-    "version": "2.2.109"
+    "version": "3.1.100"
```
Comment (brendankowitz) [inline global.json]: We'll upgrade incrementally to 3.1.101 when the runtime is available in appservice.

File: src/Microsoft.Health.Fhir.Shared.Api/Modules/MvcModule.cs [hunk 4]
```diff
@@ -40,7 +39,7 @@ public void Load(IServiceCollection services)
             services.Configure<RouteOptions>(options => options.ConstraintMap.Add(KnownRoutes.CompartmentResourceTypeRouteConstraint, typeof(CompartmentResourceTypesRouteConstraint)));
 
             // Adds provider to serve embedded razor views
-            services.Configure<RazorViewEngineOptions>(options =>
+            services.Configure<AspNetCore.Mvc.Razor.RuntimeCompilation.MvcRazorRuntimeCompilationOptions>(options =>
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Shared.Api/Modules/MvcModule.cs]: Simplify type name?

File: build/package-web.yml [hunk 5]
```diff
@@ -10,5 +10,5 @@ steps:
     inputs:
       command: publish
       projects: '${{parameters.csproj}}'
-      arguments: '--runtime win10-x64 --output $(build.artifactStagingDirectory)/web --configuration $(buildConfiguration) --version-suffix $(build.buildnumber)'
+      arguments: '--output $(build.artifactStagingDirectory)/web --configuration $(buildConfiguration) --version-suffix $(build.buildnumber)'
```
Comment (brandonpollett) [inline build/package-web.yml]: Let's verify the change by leaving this out.

@johnstairs thoughts?

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/StartupTestBase.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Api.Controllers;
+using Microsoft.Health.Fhir.Web;
+
+namespace Microsoft.Health.Fhir.Shared.Tests.E2E
+{
+    public class StartupTestBase : Startup
```
Comment (brandonpollett) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/StartupTestBase.cs]: Maybe name this something around `StartupBaseForCustomProviders`

**Pull request merged**